src/hotspot/share/gc/shenandoah/shenandoah_globals.hpp

```diff
@@ -518,6 +518,9 @@
   product(bool, ShenandoahIUBarrier, false, DIAGNOSTIC,                     \
           "Turn on/off I-U barriers barriers in Shenandoah")                \
                                                                             \
+  product(bool, ShenandoahCardBarrier, false, DIAGNOSTIC,                   \
+          "Turn on/off card-marking post-write barrier in Shenandoah")      \
+                                                                            \
```

Comment:
I was thinking this would not be visible on the command line.  If we make it visible, then we would need to assure consistency with mode == generational.

My thought was we would just set this global variable in, for example, ShenandoahArguments::initialize() if !strcmp(ShenandoahGCMode, "generational")

Reply:
This approach has parity with the other barrier flags. `shenandoahGenerationalMode.cpp` asserts that the flag is on. 

It shouldn't be strictly necessary with the default being `false`, but we may want to have `shenandoahPassiveMode.cpp` explicitly turn this flag off as it does the other barrier flags.

Reply:
If we make this visible on the command line, we will want to make sure it is consistent with mode->is_generational().  Right?

Reply:
(or you could say, if they turn off ShenandoahCardBarrier, that would mean we have no remembered set and always scan all of old-gen.  but that would be a lot more work for us, and of no value, I think...)

Reply:
maybe just fatal out in ShenandoahArguments if mode->is_generational() and !ShenandoahCardBarrier.  So the user has to specify ShenandoahCardBarrier if they specify ShenandoahGCMode == generational.

Reply:
I'm going to leave it visible on command-line like the other `Shenandoah*Barrier` flags, but force it off in all modes except generational where we force it on. Since these aren't manageable flags,, jcmd should not be able to enable them after start up either. 

If the concern is that we would be silently ignoring user request to explicitly enable it in non-generational mode, or disable it in generational mode, we can log a warning upon such an attempt and still do the right thing.

Let me test these various flag combos and ensure that they do the right thing. I'll update the PR once finished w/the changes and testing.

Reply:
Maybe the comment can say: "ShenandoahCardBarrier" is true by default iff ShenandoahGCMode is generational.

Otherwise, I think we would need to change all of the existing test configurations that have generational mode to add -XX:ShenandoahCardBarrier=true

Reply:
I updated with an attempt to address the comments above. Take a look and see how it looks.

I'll also take this out of draft form to a regular PR.

Reply:
So we'll now exit with an appropriate error message if the user changes `ShenandoahCardBarrier` to a value that conflicts with the `ShenandoahGCMode` they selected:

Reply:
Elaborated documentation so users aren't confused when they try and toggle settings to those that might conflict, and the JVM exits with a somewhat terse error message such as above.